cpp/examples/mcap/src/main.cpp

```diff
@@ -0,0 +1,31 @@
+#include <foxglove/channel.hpp>
+#include <foxglove/mcap.hpp>
+
+int main(int argc, const char* argv[]) {
+  foxglove::McapWriterOptions options = {};
+  options.path = "test.mcap";
+  options.create = true;
+  options.truncate = true;
+  foxglove::McapWriter writer(options);
+
+  foxglove::Schema schema;
+  schema.name = "Test";
+  schema.encoding = "jsonschema";
+  std::string schemaData = R"({
+    "type": "object",
+    "properties": {
+        "val": { "type": "number" }
+    }
+    })";
+  schema.data = reinterpret_cast<const std::byte*>(schemaData.data());
+  schema.dataLen = schemaData.size();
+  foxglove::Channel channel{"example", "json", std::move(schema)};
+
+  for (int i = 0; i < 100; ++i) {
+    std::string msg = "{\"val\": " + std::to_string(i) + "}";
+    channel.log(reinterpret_cast<const std::byte*>(msg.data()), msg.size());
+  }
+
+  // Optional, if you want to check for or handle errors
```

Comment:
nit: this doesn't actually let you check for errors yet 😅   soon we will figure out error handling....

Reply:
Yeah, we'll have to think how we want to do that